### PR TITLE
feat(integrations): add integration details drawer

### PR DESCRIPTION
**Pull request by InsaneZein**
### Description
This adds the table drawer for the new integrations table with the "Integration details" tab completed, and an action menu. The action menu is only visually completed. The API is not yet ready. The "Associated event types" tab will be done in a future story. 

[RHCLOUD-35060](https://issues.redhat.com/browse/RHCLOUD-35060)

---

### Screenshots
drawer
image

action menu
image

---

### Checklist ☑️
- [ ] PR only fixes one issue or story 
- [ ] Change reviewed for extraneous code 
- [ ] UI best practices adhered to 
- [ ] Commits squashed and meaningfully named 
- [ ] All PR checks pass locally (build, lint, test, E2E)

##
- [ ] _(Optional) QE: Needs QE attention (OUIA changed, perceived impact to tests, no test coverage)_
- [ ] _(Optional) QE: Has been mentioned_
- [ ] _(Optional) UX: Needs UX attention (end user UX modified, missing designs)_
- [ ] _(Optional) UX: Has been mentioned_
##

### CODE REVIEW

File: src/components/Integrations/IntegrationsDrawer.tsx [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  useDataViewEventsContext,
+} from '@patternfly/react-data-view';
+import {
+  Dropdown,
+  DropdownItem,
+  DropdownList,
+  MenuToggle,
+  MenuToggleElement,
+} from '@patternfly/react-core';
+import EllipsisVIcon from '@patternfly/react-icons/dist/esm/icons/ellipsis-v-icon';
```
Comment (karelhala) [inline src/components/Integrations/IntegrationsDrawer.tsx]:
```suggestion
import EllipsisVIcon from '@patternfly/react-icons/dist/dynamic/icons/ellipsis-v-icon';
```

**Comment by karelhala** [comment]
@InsaneZein from the failed unit tests it looks like we need to mock crypto library.

File: src/components/Integrations/IntegrationDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+import { IntegrationRow } from './Table';
+import Config, { defaultIconList } from '../../config/Config';
+import messages from '../../properties/DefinedMessages';
+import { useIntl } from 'react-intl';
+
+export const getIntegrationIcon = (type: string): React.ReactElement | null => {
+  const allIcons = Object.assign(
+    defaultIconList['Communications'],
+    defaultIconList['Reporting']
+  );
```
Comment (karelhala) [inline src/components/Integrations/IntegrationDetails.tsx]: No need to use Object.assign
```suggestion
  const allIcons = {
    ...defaultIconList['Communications'],
    ...defaultIconList['Reporting']
  };
```

File: src/components/Integrations/IntegrationDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+interface IntegrationDetailsProps {
+  integration?: IntegrationRow;
+}
+
+const IntegrationDetails: React.FunctionComponent<IntegrationDetailsProps> = ({
+  integration,
+}) => {
+  const intl = useIntl();
+
+  const buildIntegrationDetails = (integration: IntegrationRow) => {
```
Comment (karelhala) [inline src/components/Integrations/IntegrationDetails.tsx:52]: Can we break this into separate components? That way when new field needs to be added we can just adjust the component.

File: src/components/Integrations/IntegrationsDrawer.tsx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+      )}
+      shouldFocusToggleOnSelect
+    >
+      <DropdownList>
+        <DropdownItem
+          value={0}
+          key="action"
+          description={intl.formatMessage(messages.pauseDescription)}
+        >
+          {intl.formatMessage(messages.integrationdropdownPause)}
```
Comment (karelhala) [inline src/components/Integrations/IntegrationsDrawer.tsx:67]: We should toggle between pause and `enabled`. If the integration is paused we should show to cusomers option to enable this integration.

**Review by karelhala — COMMENTED** [review]
Looking good! Just a few adjustments and we are golden.

As for the APIs I've checked https://console.redhat.com/docs/api/integrations and looks like all actions are available:
* pause - `https://console.redhat.com/docs/api/integrations/v1#operations-default-EndpointResource\$V1_disableEndpoint`
* enable - `https://console.redhat.com/docs/api/integrations/v1#operations-default-EndpointResource\$V1_enableEndpoint`
* remove - `https://console.redhat.com/docs/api/integrations/v1#operations-default-EndpointResource\$V1_deleteEndpoint`
* edit - show the edit wizard

When I checked the UI and switched from one tab to another the drawer was still opened, can we close it if user clicks outside of the drawer?

If we want to show the drawer only on name click can we make it active? For instance show it in different color and pointer cursor.

**Comment by InsaneZein** [comment]
> @InsaneZein from the failed unit tests it looks like we need to mock crypto library.

~~I ended up fixing the unit tests by changing the icon import~~ ah nevermind I'm seeing the crypto failure now. Is there anywhere else in the UI we mock crypto library?

**Comment by InsaneZein** [comment]
> As for the APIs I've checked https://console.redhat.com/docs/api/integrations and looks like all actions are available:

I can tackle this in a different PR and ticket since this ticket assumed actions api was not available.

**Comment by InsaneZein** [comment]
@karelhala I fixed the table so that it shows the focused row when the drawer is opened, and changes the cursor to show the whole row is clickable.

File: src/components/Integrations/IntegrationsDrawer.tsx [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+        <Tab
+          eventKey={1}
+          title={
+            <TabTitleText>
+              {intl.formatMessage(messages.associatedEventTypesTabTitle)}
+            </TabTitleText>
+          }
+          ouiaId={`${ouiaId}-associated-event-types-tab`}
+        >
+          hi
```
Comment (karelhala) [inline src/components/Integrations/IntegrationsDrawer.tsx]: No need to show debug values in columns, let's add either null or empty string if we don't have the data available.

**Review by karelhala — COMMENTED** [review]
Looking good! Just one small debug fix and we are golden.

File: src/components/Integrations/IntegrationsDrawer.tsx (hunk 5 above)
Comment (InsaneZein) [reply src/components/Integrations/IntegrationsDrawer.tsx]: ok I took it out

**Review by karelhala — APPROVED** [review]

**Pull request merged**